src/main/java/org/opensearch/security/support/SecurityUtils.java

```diff
@@ -140,4 +150,53 @@ private static String resolveEnvVar(String envVarName, String mode, boolean bc,
             return bc ? Hasher.hash(envVarValue.toCharArray(), settings) : envVarValue;
         }
     }
+
+    // This is used to match host names - for e.g. *.example.local to dev.example.local
+    public static boolean matchesHostNamePatterns(WildcardMatcher hostMatcher, InetAddress address, String hostResolverMode) {
+        if (address == null || hostMatcher == null) {
+            return false;
+        }
+
+        // Only proceed with hostname checks if hostResolverMode is configured
+        if (hostResolverMode != null
+            && (hostResolverMode.equalsIgnoreCase("ip-hostname") || hostResolverMode.equalsIgnoreCase("ip-hostname-lookup"))) {
+            try {
+                List<String> valuesToCheck = new ArrayList<>(List.of(address.getHostAddress()));
+                final String hostName = address.getHostName();
+                valuesToCheck.add(hostName);
+                return valuesToCheck.stream().anyMatch(hostMatcher);
+            } catch (Exception e) {
+                log.warn("Failed to resolve hostname for {}: {}", address.getHostAddress(), e.getMessage());
+                return false;
+            }
+        }
+        return false;
+    }
+
+    // This is used to match IP addresses - for e.g. 192.168.0.1 to 192.168.0.0/16
+    public static boolean matchesIpAndCidrPatterns(ClientBlockRegistry<InetAddress> clientBlockRegistry, InetAddress address) {
+        if (address == null || clientBlockRegistry == null) {
+            return false;
+        }
+
+        AuthFailureListener authFailureListener = (AuthFailureListener) clientBlockRegistry;
```

Comment:
Nit: As this method only works for `AuthFailureListener` instances, it should be probably an instance method of `AuthFailureListener`. 

Alternatively: Have you considered to create a new class that encapsulates a combination of `WildcardMatcher` and CIDR matching? 

This combination of matching is something what is not only useful for the `ignore_hosts` setting, but for quite a few functionalities. Having a reusable component for that would be nice.

Reply:
makes sense. will refactor the changes